torchao/prototype/smoothquant/example.py

```diff
@@ -4,185 +4,276 @@
+            nlls.append(neg_log_likelihood)
+
+            prev_end_loc = end_loc
+            if end_loc == seq_len:
+                break
+
+    ppl = torch.exp(torch.stack(nlls).sum() / end_loc)
+
+    return {
+        "perplexity": ppl.item(),
+        "tokens_per_sec": input_ids.size(1) / inference_time,
+    }
+
+
+def quantize_and_eval(
+    repo_id: str,
+    alpha: float,
+    tasks: list[str],
+    max_seq_length: int,
+    calibration_limit: int,
     device: str,
     precision: torch.dtype,
-    sequence_length: int,
     compile: bool,
-    model_load_path: str,
     model_save_path: str,
+    model_save_hf_hub_path: str,
 ):
     print(f"Loading model on {device}...")
     torch.manual_seed(34)
     t0 = time.time()
-    tokenizer = AutoTokenizer.from_pretrained(model_id)
-    if model_load_path is not None and os.path.exists(model_load_path):
-        print(f"Loading quantized model from {model_load_path}")
-        t0 = time.time()
-        model = torch.load(model_load_path, weights_only=False).to(device)
-        print(f"Time to load quantized model: {time.time() - t0:.02f} seconds")
-    else:
-        model = (
-            AutoModelForCausalLM.from_pretrained(model_id, torch_dtype=precision)
-            .eval()
-            .to(device)
-        )
-        print(f"Time to load model: {time.time() - t0:.02f} seconds")
-        print("running calibration")
-        t0 = time.time()
-        # insert observers to find average magnitude and calculate scales
-        insert_smooth_quant_observer_(model, alpha, quant_mode)
-        calibration_data = get_calib_dataset(
-            tokenizer=tokenizer, n_samples=calibration_size, block_size=sequence_length
-        )
-        for batch in calibration_data:
-            model(batch.to(device))
-            batch.to("cpu")
-        print(f"time for calibration: {time.time() - t0:.02f} seconds")
-
-        is_observed_linear = lambda m, fqn: isinstance(m, SmoothQuantObservedLinear)
-        print(f"running SmoothQuant with {quant_mode} quantization")
-        t0 = time.time()
-        quantize_(model, SmoothQuantConfig(), is_observed_linear)
-        print(f"time for quantization: {time.time() - t0:.02f} seconds")
-        if model_save_path is not None:
-            print(f"Saving quantized model to {model_save_path}")
-            t0 = time.time()
-            torch.save(model, model_save_path)
-            print(f"Time to save quantized model: {time.time() - t0:.02f} seconds")
+    tokenizer = AutoTokenizer.from_pretrained(repo_id)
+    model = (
+        AutoModelForCausalLM.from_pretrained(repo_id, torch_dtype=precision)
+        .eval()
+        .to(device)
+    )
+    print(f"Time to load model: {time.time() - t0:.02f} seconds")
+
+    # Step 1: Prepare - insert observers
+    print("running SmoothQuant prepare and calibrate")
+    t0 = time.time()
+    quant_config = SmoothQuantConfig(
+        base_config=Int8DynamicActivationInt8WeightConfig(),
```

Comment:
so this is int8 dynamic activation + int8 weight? instead of int8 + int4? did you use a different one when you run the test?